packages/flutter_tools/lib/src/resident_web_runner.dart

```diff
@@ -179,27 +221,80 @@ class ResidentWebRunner extends ResidentRunner {
       timeout: timeoutConfiguration.fastOperation,
       progressId: 'hot.restart',
     );
-    OperationResult result = OperationResult.ok;
-    try {
-      final List<Uri> invalidatedSources = ProjectFileInvalidator.findInvalidated(
-        lastCompiled: _lastCompiled,
-        urisToMonitor: <Uri>[
-          for (FileSystemEntity entity in flutterProject.directory
-              .childDirectory('lib')
-              .listSync(recursive: true))
-            if (entity is File && entity.path.endsWith('.dart')) entity.uri
-        ], // Add new class to track this for web.
-        packagesPath: PackageMap.globalPackagesPath,
-      );
-      await webCompilationProxy.invalidate(inputs: invalidatedSources);
-      await _connection.sendCommand('Page.reload');
-      await Future<void>.delayed(const Duration(milliseconds: 150));
-    } catch (err) {
-      result = OperationResult(1, err.toString());
-    } finally {
-      printStatus('Restarted application in ${getElapsedAsMilliseconds(timer.elapsed)}.');
-      status.cancel();
-    }
-    return result;
+    final Response response = await vmService.callMethod('hotRestart');
+    return OperationResult.ok;
+
+    // try {
+    //   final List<Uri> invalidatedSources = ProjectFileInvalidator.findInvalidated(
+    //     lastCompiled: _lastCompiled,
+    //     urisToMonitor: <Uri>[
+    //       for (FileSystemEntity entity in flutterProject.directory
+    //           .childDirectory('lib')
+    //           .listSync(recursive: true))
+    //         if (entity is File && entity.path.endsWith('.dart')) entity.uri
+    //     ], // Add new class to track this for web.
+    //     packagesPath: PackageMap.globalPackagesPath,
+    //   );
+    //   await webCompilationProxy.invalidate(inputs: invalidatedSources);
+    //   await _connection.sendCommand('Page.reload');
+    //   await Future<void>.delayed(const Duration(milliseconds: 150));
+    // } catch (err) {
+    //   result = OperationResult(1, err.toString());
+    // } finally {
+    //   printStatus('Restarted application in ${getElapsedAsMilliseconds(timer.elapsed)}.');
+    //   status.cancel();
+    // }
+    // return result;
+  }
+
+  @override
+  Future<void> debugDumpApp() async {
+    final Response response = await vmService.callMethod('flutter.ext.debugDumpApp');
+  }
+
+  @override
+  Future<void> debugDumpRenderTree() async {
+    final Response response = await vmService.callMethod('flutter.ext.debugDumpRenderTree');
+  }
+
+  @override
+  Future<void> debugDumpLayerTree() async {
```

Comment:
Assuming this vmservice works the way I think, most of these extensions could be implemented this way (and then printing some result for the ones that produce output).

Reply:
Actually we get the stdout through the vmservice already